src/libp2p/websocketOverTor.ts

```diff
@@ -111,12 +113,28 @@ class WebsocketsOverTor extends WebSockets {
     return rawSocket
   }
 
-  prepareListener = ({ handler, upgrader }, options = {}) => {
```

Comment:
be very careful here,
I think that when libp2p runs this method it may be passing something in `options`, which is important for this to work properly.

Please verify this (by checking if the whole thing runs now and connects correctly)